### DISCUSSION TITLE
posts/2023/09/new-life-old-mac-linux-follow-up/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#56] [Announcements]
# posts/2023/09/new-life-old-mac-linux-follow-up/

Some things I’ve picked up while playing around with various aspects of Big L.

https://www.brycewray.com/posts/2023/09/new-life-old-mac-linux-follow-up/

**Comment by sleekweasel** [comment]
I was in just the same position regarding typing after I went to university, but during a postgraduate course, I noticed I knew where the keys were and just resolved to try using the home keys method. I slowed down to 40 wpm or so for a week or two, but eventually retrained myself. The only mistake I made was learning the wrong finger for 'b'.

So to clarify, if I wipe out the EFI, I can still use the R-boot to restore MacOS if everything goes wrong? And what happens if I don't use MacOS to wipe out the GUID one? 

Cheers!

**Reply by brycewray** [reply]
@sleekweasel I haven’t researched this in over a year, so have nothing new to add to what I wrote in the original. Sorry if that’s not sufficiently helpful, but I want to be honest about it. Best wishes.

**Reply by sleekweasel** [reply]
Sure, no worries. I'm about to mess with my wife's old Mac. I'll update here if anything exciting happens.

**Comment by taprobane99** [comment]
There is no magic about Apple using 4096x2304. It's the max allowed resolution for Displayport 1.2 and HDMI 2.0. Inside the iMac is a similar controller. It's just they're the only manufacturer to go that high. It can be set with CVT-RB2 video timings in linux. Add the following to your kernel command line (edit /etc/default/grub with sudo nano, then sudo update-grub)

video=eDP-1:4096x2304MR

**Reply by taprobane99** [reply]
ah oops i didn't realise you had the 5k imac. please delete above comment ^

**Comment by alfredo87** [comment]
Hi, nice write up. I have done this putting linux on thunderbolt drive and managed to get it working. Suspend,etc. doe not work.
See here what I have done: https://github.com/alfredo87/iMacEOL
A lot of ideas came from your two articles:thank you very much for writing it up and publishing it

**Reply by brycewray** [reply]
@alfredo87 Thank you for the kind words! I’m glad they helped you. This post seems to get a great deal of my site’s traffic, so it’s clear there’s a lot of Linux-on-Intel-Macs interest out there. Also: nice work on your own writeup (and the work that went into it).

**Note**: I, too, have had my issues with suspend on my old iMac with some distros I’ve tried. Am currently using Fedora 42 and the problem seems for now to have gone away, but I still don’t leave that Mac powered up overnight as I do with the daily-driver Mac running macOS.

**Reply by taprobane99** [reply]
hi alfredo. did you actually get the full 5K resolution? as far as I know nobody has managed to do that yet in Linux and the iMac only runs at 4K.

**Comment by Fred68212** [comment]
I recently put Linux on a 17" Intel iMac my last gave me, and it was quite the adventure. I replaced the HDD with a 250 gig SSD (kudos to iFixit for their almost accurate guide on how to get the box open and swap out the drive). My first attempt was using Linux Mint, 
sometimes the live USB would boot, sometimes it left me staring at a blank screen, etc. I did finally make to install it, but could not get it to boot. Even though the machine is capable of running a 64 bit OS, the EFI bootloader is 32 bit. After futzing around with it, I opened the box, pulled the drive out, plugged it into my raspberry pi and used Gparted to wipe the partitions. I next tried antiX, which installed. I had to manually run install-grub to get it to boot, and it worked well, but did not play nice with XFCE4 and firewalld, my preferred firewall. So I started all over again, removing the drive (I'm real good at it now) and wiping it. Next I tried MX Linux, which supports XFCE4, and the install went off without a hitch, the machine worked well  Then a week later they released MX25, which is based on the current version of Debian. The computer is great, even though sometimes I have to try 3 or 4 times to get it booted. The people on the MX forum weren't overly helpful with the booting issue, which is ok. After reading the docs I now know grub fairly well. Maybe I will eventually solve the booting issue, but it not it's ok, because it's my number 3 computer. I did the whole project for no other reason than to see if I could do it.

**Reply by brycewray** [reply]
@Fred68212 ... which may be the very best reason for folks like us! :-) I salute your perseverance.

**Comment by mnkygrl** [comment]
Thanks for this! Enjoyed reading it, good information and humor! Recently installed Zorin on an early '08 MacBook, dual boot setup. Thanks especially for the Text Fiends part, would've saved me time (alas I only found your article now).

**Reply by brycewray** [reply]
@mnkygrl Thanks for the kind words, and I’m glad you found it helpful — albeit not quite in time for when you needed it. :)
